scripts/build/add-basic-constructor-to-react-component.js

```diff
@@ -0,0 +1,40 @@
+// this is to avoid the spread argument helper for react component constructors
```

Comment:
could u explain what exactly is avoided with this and why this is a problem?

Reply:
https://runkit.com/mitchellhamilton/5b370eeaf5b34c0012874d92
Without this, the constructor's arguments will be iterated over and put in an array unnecessarily.

Reply:
nice! this would be useful as part of `preset-react`, gonna propose it later to rest of the babel team

Reply:
BTW, this implementation isn't perfect since it should technically pass the second argument to super as well (and maybe the third but I don't know if that's strictly necessary). The reason I didn't do that here is that I know I won't access context in those cases.

https://github.com/facebook/react/blob/master/packages/react/src/ReactBaseClasses.js#L21-L29

Reply:
no idea how `updater` arg might be used 🤔  we could do `super(props, context, updater)` by default + an option to optimize it to just `super(props)`

Not sure if `context` here is only for legacy purposes, or is the new one also passed here - anyway doesn't matter as long as this arg is here